testing/proftpd/APKBUILD

```diff
@@ -0,0 +1,147 @@
+	subpackages="$subpackages $pkgname-mod_${_module%:*}:_module"
+	_shared_mods="${_shared_mods:+$_shared_mods:}mod_${_module%:*}"
+done
+subpackages="$pkgname-dev $pkgname-doc $pkgname-utils
+	$pkgname-mod_sftp $pkgname-mod_snmp $subpackages
+	"
+source="ftp://ftp.proftpd.org/distrib/source/$pkgname-$_pkgver.tar.gz
+	$pkgname.logrotate
+	$pkgname.initd
+	mod_delay.conf
+	mod_load.conf
+	mod_sftp.conf
+	mod_tls_shmcache.conf
+	libressl.patch
+	"
+builddir="$srcdir/$pkgname-$_pkgver"
+
+build() {
+	cd "$builddir"
+	./configure \
+		--prefix=/usr \
+		--sysconfdir=/etc/$pkgname \
+		--localstatedir=/run/$pkgname \
+		--libexecdir=/usr/lib/$pkgname \
+		--disable-strip \
+		--disable-pam \
+		--enable-dso \
+		--enable-facl \
+		--enable-pcre \
+		--enable-ipv6 \
+		--enable-ctrls \
+		--enable-openssl \
+		--with-shared=mod_sftp:mod_snmp:$_shared_mods \
+		|| return 1
+	make || return 1
+}
+
+package() {
+	cd "$builddir"
+	make DESTDIR="$pkgdir" install || return 1
+	sed -i	-e "5 a Include /etc/$pkgname/modules.d/*.conf" \
+		-e "5 a Include /etc/$pkgname/conf.d/*.conf\n" \
+		-e "s/nobody/$pkgusers/" \
+		-e "s/nogroup/$pkggroups/" \
+		"$pkgdir"/etc/$pkgname/$pkgname.conf || return 1
+	install -d "$pkgdir"/var/log/$pkgname
+	install -d "$pkgdir"/etc/$pkgname/conf.d
+	install -d "$pkgdir"/etc/$pkgname/modules.d
+	install -dm750 -o ftp -g ftp "$pkgdir"/var/lib/ftp
+	install -Dm755 "$srcdir"/$pkgname.initd "$pkgdir"/etc/init.d/$pkgname
+	install -Dm644 "$srcdir"/$pkgname.logrotate "$pkgdir"/etc/logrotate.d/$pkgname
+	rm -fr "$pkgdir"/run "$pkgdir"/usr/share/locale \
+		"$pkgdir"/usr/sbin/in.$pkgname
+}
+
+dev() {
+	default_dev
+	mkdir -p "$subpkgdir"/usr/bin
+	mv "$pkgdir"/usr/bin/prxs "$subpkgdir"/usr/bin || return 1
+}
+
+utils() {
+	pkgdesc="$pkgdesc (utilities)"
+	depends="$pkgname perl perl-mail-sendmail"
+	mkdir -p "$subpkgdir"/usr/sbin
+	mv "$pkgdir"/usr/bin "$subpkgdir"/usr || return 1
+	mv "$pkgdir"/usr/sbin/ftpscrub \
+		"$pkgdir"/usr/sbin/ftpshut \
+		"$subpkgdir"/usr/sbin || return 1
+}
+
+_module() {
+	local name=${subpkgname#$pkgname-}
+	local load=$(eval echo \$_load_$name)
+	pkgdesc="ProFTPd module: $name"
+	depends="$pkgname"
+	mkdir -p "$subpkgdir"/usr/lib/$pkgname
+	mv "$pkgdir"/usr/lib/$pkgname/$name.so \
+		"$subpkgdir"/usr/lib/$pkgname || return 1
+	[ -f "$srcdir"/$name.conf ] && install -Dm644 "$srcdir"/$name.conf \
+		"$subpkgdir"/etc/$pkgname/conf.d/${name#mod_}.conf
+	if [ "$load" != "--" ]; then
+		local file="$subpkgdir"/etc/$pkgname/modules.d/${name#mod_}.conf
+		if [ -z "${load##=*}" ]; then
+			file="$subpkgdir"/etc/$pkgname/modules.d/${load#=}${name#mod_}.conf
+			load=""
+		fi
+		mkdir -p ${file%/*}
+		echo "<IfModule !$name.c>" > $file
+		local dep; for dep in $(echo $load | tr "," " "); do
+			depends="$depends $pkgname-mod_$dep"
+			echo "  <IfModule !mod_$dep.c>" >> $file
+			echo "    LoadModule mod_$dep.c" >> $file
+			echo "  </IfModule>" >> $file
+		done
+		echo -e "  LoadModule $name.c\n</IfModule>" >> $file
```

Comment:
Please don’t use `echo -e`, its behaviour is not well defined. Use `printf` instead.